pkg/controller/elasticsearch/client/shard.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package client
+
+import (
+	"context"
+)
+
+// AllocationSetter captures Elasticsearch API calls around allocation filtering.
+type AllocationSetter interface {
+	ExcludeFromShardAllocation(nodes string) error
+}
+
+// ShardLister captures Elasticsearch API calls around shards retrieval.
+type ShardLister interface {
+	GetShards() (Shards, error)
+}
+
+type clientWrapper struct {
```

Comment:
Can you help me understand why we need the wrapper?

Reply:
Because I need an object to implement the high-level interfaces _(that being said, as I still feel new to Go maybe I'm missing the obvious :wink: )_

Reply:
Since we are wrapping a struct in the same package I wasn't sure if it made sense to just add those funcs to Client

Reply:
My idea was to have something easy to mock, bring an abstraction over the current HTTP client without introducing some additional complexity to it.

Reply:
👍 , I will update this part with a new PR